### DISCUSSION TITLE
working on universalizing a personal fork - halp?

### DISCUSSION THREAD

**Discussion by shamelesslymkp** [nianeyna/ao3downloader#46] [Q&A] [answered]
Hi, @nianeyna! tumblr user shamelesslymkp here, trying to puzzle out how to automate the generation of user-specific AO3 urls based on the username provided for ao3 login - I feel like this should be possible to do, maybe with something like a string template (see below), but I'm not sure how to reference the username variable. Can you assist?

```
s = Template ('$ao3$username$url')
s.substitute(ao3='https://www.archiveofourown.org/', username='?', url='/readings?show=to-read')
```

**Comment by nianeyna** [comment, marked as answer]
You can use the methods in fileio to interact with the settings file. You can also use f-strings as a simpler way of doing string formatting - here's an example

```py
import ao3downloader.fileio as fileio
import ao3downloader.strings as strings

username = fileio.get_setting(
    strings.SETTINGS_FILE_NAME, 
    strings.SETTING_USERNAME)

url = f'https://archiveofourown.org/users/{username}/readings?show=to-read'
```

if you want to prompt the user for their username in the case that it's not already saved in settings, you can use the `fileio.setting` method instead

**Reply by shamelesslymkp** [reply]
oh bless you! I knew it had to be somewhere xD
